Probesys/bileto
# new: Allow to login using LDAP

## Related issue(s)



https://github.com/Probesys/bileto/issues/168

## Changes



- add a ldap service to the docker-compose file
- add the symfony/ldap component and configure it with env vars
- add a ldapIdentifier to users
- create and configure a custom authenticator to handle local and ldap auth
- disallow the edition of profiles managed by LDAP

## How to test manually



- start the dev env with `make docker-run LDAP=true`
- login with `alix@example.com / secret` → check it works
- login with `charlie@example.com / secret` → check it **does not** work
- login with `charlie / secret` → check it works
- check in DB that the user `dominique@example.com` does not work
- login with `dominique / secret` → check it works and that the user `dominique@example.com` now exists in DB
- go to your profile → check the user is named "Dominique Aragua" but that you can't change your information

## Checklist
